### PR TITLE
Add github actions workflow

### PR DESCRIPTION
**Pull request by tkoskela**
Adds a GitHub actions workflow for running cmake and ctest, similar to https://github.com/astro-informatics/purify/pull/299

### CODE REVIEW

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        path: ${{runner.workspace}}/.ccache
+        key: ${{matrix.os}}-${{matrix.cxx}}-${{matrix.mpi}}-${{matrix.omp}}-${{ steps.ccache_cache_timestamp.outputs.timestamp }}
+        restore-keys: |
+          ${{ matrix.os }}-${{ matrix.cxx }}-${{ matrix.mpi }}-${{ matrix.omp }}
+          ${{ matrix.os }}-${{ matrix.cxx }}-${{ matrix.mpi }}
+          ${{ matrix.os }}-${{ matrix.cxx }}
+          ${{ matrix.os }}
+
+#    - name: Clear ccache
+#      run: ccache --clear
```
Comment (SJaffa) [inline .github/workflows/cmake.yml]: If this commented out section is never needed should we delete it? If it is only used in certain circumstances could it be conditional like the OS specific ones?

File: .github/workflows/cmake.yml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      # See https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html?highlight=cmake_build_type
+      run: |
+        cmake -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}} \
+        -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
+        -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON \
+        -Ddompi=${{matrix.mpi}} -Dopenmp=${{matrix.omp}}
+
+    - name: Build
+      # Build your program with the given configuration
+      run: cmake --build ${{github.workspace}}/build --config ${{env.BUILD_TYPE}} --parallel 2
```
Comment (SJaffa) [inline .github/workflows/cmake.yml:96]: Could you add a comment to explain why you use `--parallel 2`?

File: cmake_files/dependencies.cmake [hunk 3]
```diff
@@ -48,14 +48,14 @@ if(openmp)
     set_target_properties(openmp::openmp PROPERTIES
       INTERFACE_COMPILE_OPTIONS "${OpenMP_CXX_FLAGS}"
       INTERFACE_LINK_LIBRARIES  "${OpenMP_CXX_FLAGS}")
-  else()
+  else()
```
Comment (SJaffa) [inline cmake_files/dependencies.cmake]: It's a bit hard to see on the GitHub interface but is this edit just whitespace after the `else()`? If so it should probably be removed?

File: .github/workflows/cmake.yml (hunk 1 above)
Comment (tkoskela) [reply .github/workflows/cmake.yml]: I think we can delete it. I tried it when we were suspecting that `gcc` was pulling wrong header files from the cache but that was not the case.

File: .github/workflows/cmake.yml (hunk 2 above)
Comment (tkoskela) [reply .github/workflows/cmake.yml:96]: Updated the comment

File: cmake_files/dependencies.cmake (hunk 3 above)
Comment (tkoskela) [reply cmake_files/dependencies.cmake]: Yep, I keep forgetting to run `M-x delete-trailing-whitespace`

**Review by SJaffa — APPROVED** [review]

**Pull request merged**